src/api/resources/empathicVoice/resources/chat/client/Client.ts

```diff
@@ -0,0 +1,47 @@
+import * as core from "../../../../../../core";
+import qs from "qs";
+import { ChatSocket } from "./Socket";
+
+export declare namespace Chat {
+    interface Options {
+        apiKey?: core.Supplier<string | undefined>;
+        accessToken?: core.Supplier<string | undefined>;
+    }
+
+    interface ConnectArgs {
+        /** The ID of the configuration. */
+        configId?: string;
+
+        /** The version of the configuration. */
+        configVersion?: string;
+
+        /** The ID of a chat group, used to resume a previous chat. */
+        resumedChatGroupId?: string;
+    }
+}
+
+export class Chat {
+    constructor(protected readonly _options: Chat.Options) {}
+
+    public connect(args: Chat.ConnectArgs = {}): ChatSocket {
+        const queryParams: Record<string, string | string[] | object | object[]> = {};
+
+        queryParams["accessToken"] = core.Supplier.get(this._options.accessToken);
```

Comment:
it would be useful if we only allowed one of these to be appended to the URL

https://github.com/HumeAI/empathic-voice-api-js/blob/3d665373ee99d8edf3b849d2a82b7c782fa5bb13/packages/core/src/lib/create-url.ts#L20